sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FilterPushdownSuite.scala

```diff
@@ -1016,6 +1016,8 @@ class FilterPushdownSuite extends PlanTest {
     val correctAnswer = x.where("x.a".attr === 5).join(y.where("y.a".attr === 5),
         condition = Some("x.a".attr === Rand(10) && "y.b".attr === 5))
 
-    comparePlans(Optimize.execute(originalQuery.analyze), correctAnswer.analyze)
```

Comment:
```
db2 "select a.actno from act a, act b where a.actno = b.actno and rand() < 0.5 and rand() >= 0.5"
```

In the above example, the non-deterministic functions in the join conditions are also evaluated twice in DBMS.